src/reuse/modules/mobile/device.ts

```diff
@@ -0,0 +1,116 @@
+"use strict";
+
+import { VerboseLoggerFactory } from "../../helper/verboseLogger";
+import ErrorHandler from "../../helper/errorHandler";
+
+/**
+ * @class device
+ * @memberof Mobile
+ */
+export class Device {
+  private vlf = new VerboseLoggerFactory("mobile", "device");
+  private ErrorHandler = new ErrorHandler();
+
+  /**
+   * @function isAppInstalled
+   * @memberOf mobile.device
+   * @description Check wether given package/bundle app is installed or not in the device.
+   * @param {string} packageIdorBundleId - Android package Id, or iOS bundle Id.
+   * @returns {boolean} Returns true if specified app package/bundled installed in the device, or false.
+   * @example await mobile.device.isAppInstalled("com.google.android.apps.maps");
+   */
+  async isAppInstalled(packageIdorBundleId: string): Promise<boolean> {
+    const vl = this.vlf.initLog(this.isAppInstalled);
+    try {
+      const isAppInstalledinDevice: boolean = browser.isAppInstalled(packageIdorBundleId);
+      vl.log(`Given app package/bundle id ${packageIdorBundleId} installed on the device is ${isAppInstalledinDevice.toString()}`);
+      return isAppInstalledinDevice;
+    } catch (error) {
+      return this.ErrorHandler.logException(error, 'Error: During isAppInstalled', true);
+    }
+  }
+
+/**
+   * @function installAppForPlatform
+   * @memberOf mobile.device
+   * @description Install the appropriate app based on the platform the test is being executed on.
+   * @param {string} appPath - Path of the app(.apk, .ipa)
+   * @example 
+   * await mobile.device.installAppForPlatform("/path/to/your/app.apk");
+   * await mobile.device.installAppForPlatform("/path/to/your/app.ipa");
+   */
+async installAppForPlatform(appPath: string): Promise<void> {
+  const vl = this.vlf.initLog(this.installAppForPlatform);
+  const platform = await browser.capabilities.platformName;
+  try {
+      if (platform.toLowerCase() === 'android') {
```

Comment:
The code in each block of the if-else statement is exactly the same, do you really need that if-else statement to check for the platform? And also, since the installation logic is the same for each platform, you might also need to name the method as just `installApp`

Reply:
agree, would put every check in one single if statement if any of the platforms are matched -> if so, install the app on the supported plattform and print a message that it was successfully installed on the given platform.

Reply:
you can dynamically set the log using the platform variable and avoid the differentiation as the above comments already explained.

Reply:
Done, Added logic with single if condition.